### PR TITLE
[feature] Implement script that plays player dies audio from GM

### LINKED ISSUE #166: Implement script that plays player dies audio from GM
- Script is called PlayerStates attached in all Player prefabs.
- Use observer pattern to notify GM
- Add AudioManager to GM and a function that will play that Audio

### LINKED ISSUE #179: Implement SFX player footstep
- pastiin ga berisik kalo ada 4 orang jalan deketan
- pastiin distancenya pas

### LINKED ISSUE #180: Implement SFX ankh

### PR DESCRIPTION
**Pull request by RaiRiz**
the script implemented is still a dummy class so currently it can't be tested

### Main changes

- Refactor AudioManager.cs filename to AudioPlayer.cs. Then, create an actual but dummy AudioManager class and prefab. A player dies audio event has been prepared in GameManager class. Need to wait for PlayerStates.cs to be implemented first for future uses.
- Add new sfx for player walk, player run, and ankh activation/deactivation. The place where players put their feet is assumed to be a ceramic surface.
- Each of player walk and player run sfx is divided into 2 types namely 1 step and 2 steps. Both can be used in combination, separately, or with loop according to needs.

### Tasks List

- [close #166 ]
- [close #179 ]
- [close #180 ]

### Attachments

- Refactor filename: AudioManager.cs to AudioPlayer.cs
image

- Actual but dummy AudioManager
image

- Player walk sfx files
image

- Player walk 1 step sfx on play
https://discord.com/channels/918510403909214268/918514897422258216/942359896995795014

- Player walk 2 steps sfx on play
https://discord.com/channels/918510403909214268/918514897422258216/942359182592581702

- Player run sfx files
image

- Player run sfx 1 step on play
https://discord.com/channels/918510403909214268/918514897422258216/942367215557169173

- Player run 2 steps sfx on play
https://discord.com/channels/918510403909214268/918514897422258216/942367194380111914

- Ankh sfx
image

- Ankh sfx on play
https://discord.com/channels/918510403909214268/918514897422258216/941224093175390239

- Player footstep with right foot first
https://discord.com/channels/918510403909214268/918514897422258216/942433205615747144

### CODE REVIEW

File: Assets/Astralization/Scripts/Managers/AudioManager.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class AudioManager : MonoBehaviour
```
Comment (uciii) [inline Assets/Astralization/Scripts/Managers/AudioManager.cs:5]: class docs

**Review by uciii — CHANGES_REQUESTED** [review]
han saran, possible gk nambah suara yang satu langkah kakinya di lari sama jalan? jadi kalo suara yang 1 langkah ini suara kak kiri, tambahin suara kaki kanannya. Biar kalo jadi pake langkah kaki yang satu satu bisa beda kanan kirinya?

atau menurut lu oke 1 suara aja, jadi kiri kanannya suaranya sama? feel free to challenge  this suggestion.

File: Assets/Astralization/Scripts/Managers/AudioManager.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+{
+    [Space]
+    [Header("Audio")]
+    [Tooltip("Audio Manager")]
+    private AudioPlayer AudioPlayerObj;
```
[the remark this answers is not in the document]
Comment (RaiRiz) [reply Assets/Astralization/Scripts/Managers/AudioManager.cs:10]: oke, akan dieksekusi segera

**Comment by RaiRiz** [comment]
sebetulnya 1 ragam suara yg sama utk kaki kiri dan kanan nggak apa. tapi betul sih, ada baiknya dibedakan. lebih masuk ke logika auditori-nya sekaligus membantu memberitahu pemain bahwa memang avatarnya sedang berjalan/berlari dengan melangkahkan kaki yang berbeda bergantian dan bukan selangkah2 kaki yg sama terus. akan dieksekusi segera

File: Assets/Astralization/Scripts/Managers/AudioManager.cs (hunk 1 above)
Comment (RaiRiz) [reply Assets/Astralization/Scripts/Managers/AudioManager.cs:5]: yang ini gimana maksudnya?

**Review by raihansyahdean — CHANGES_REQUESTED** [review]
Rename audio files:
instead of 1Step better use OneStep, 2Step better TwoSteps, etc.

File: Assets/Astralization/Scripts/Managers/AudioManager.cs (hunk 1 above)
Comment (GustiNgurahYama) [reply Assets/Astralization/Scripts/Managers/AudioManager.cs:5]: bikin docs buat classnya di codenya han, jadi diatas kelas audiomanager ada semacam
```
/*
 * Class to manage bla bla bla
 */
```

File: Assets/Astralization/Scripts/Managers/GameManager.cs [hunk 3]
```diff
@@ -25,6 +25,7 @@ private enum _playerState
     public static event Action<bool> ChangeWorldEvent;
     public static event Action<string> PlayerActionMapEvent;
     public static event Action<bool> ShowDialogueHudEvent;
+    public static event Action PlayerAudioDiesEvent;
```
[the remark this answers is not in the document]
Comment (RaiRiz) [reply Assets/Astralization/Scripts/Managers/GameManager.cs:28]: oke, akan dieksekusi segera

**Review by raihansyahdean — APPROVED** [review]
Nice!

**Review by VadimBhirawa — APPROVED** [review]
Sounds good!

**Review by uciii — APPROVED** [review]
LGTM

**Pull request merged**